bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/ui/QuickSearchDialog.java

```diff
@@ -1268,6 +1268,9 @@ protected QuickTextQuery createFilter() {
 	 */
 	protected void applyFilter(boolean force) {
 		QuickTextQuery newFilter = createFilter();
+		if(newFilter != null) {
+			getShell().setText(newFilter.getPatternString());
```

Comment:
I don't think the pattern string should replace the title.   The new filter can't be null and the pattern string of that filter is the same as pattern.getText().

Reply:
<img width="602" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.ui/assets/122080218/a446e9a7-2e73-4e5b-b17a-52494c21a7bb">

@merks I removed the null check and used `pattern.getText` instead of `getPatternString`. However I don't know what is the benefit of one over the other. I am able to get the title changed in both the ways.